datastore.go

```diff
@@ -79,15 +79,25 @@ var DefaultOptions Options
 
 func init() {
 	DefaultOptions = Options{
-		GcDiscardRatio: 0.2,
+		GcDiscardRatio: 0.01,
 		GcInterval:     15 * time.Minute,
 		GcSleep:        10 * time.Second,
 		Options:        badger.LSMOnlyOptions(""),
 	}
-	// This is to optimize the database on close so it can be opened
-	// read-only and efficiently queried. We don't do that and hanging on
-	// stop isn't nice.
-	DefaultOptions.Options.CompactL0OnClose = false
+	// This is to optimize the database on closure
+	DefaultOptions.Options.CompactL0OnClose = true
+	
+	// Remove elements which the has been deleted from the database
+	DefaultOptions.Options.NumVersionsToKeep = 0
```

Comment:
@jsign this seems wrong. Shouldn't the default of 1 keep one version? From my reading of the code, 0 will just be buggy.

Reply:
Well, if we delete a CID, it will remain in the database, since a delete will update the entry that it was deleted - not actually delete the entry.

You need to set it to `0` to let the badger gc clear up deleted entries.

That's what @jarifibrahim wrote [here](https://github.com/dgraph-io/badger/issues/1297#issuecomment-612770631):

> Also, please note that the default value of NumVersionsToKeep is 1 which means your deleted entries are also stored in the DB. You might want to set it to 0 if you don't want any deleted/expired keys to be stored in the DB.

Reply:
@RubenKelevra @Stebalien please do not set it to 0. I made a mistake. I'll update my comment on the original issue. Apologies for the confusion.